docker/README.md

```diff
@@ -0,0 +1,84 @@
+# Install CAMEL with Docker
+
+Docker offers an easy way to create a consistent and isolated virtual
+environment, containers, for setting up the dependencies of CAMEL. This guide
+will show you how to quickly set up CAMEL, run the examples, and also
+develop on it, with Docker.
+
+## Prerequisites
+- Docker：https://docs.docker.com/engine/install/
+- Docker Compose：https://docs.docker.com/compose/install/
+
+## Build Image
+To build the image of CAMEL, first you need to navigate into the
+[docker](../docker) folder and create a `.env` file with your API keys.
+
+```bash
+cd docker
+
+# Create a .env file with your OpenAI API key only
+# For other optional keys that can be added into .env, see .env.example
+echo OPENAI_API_KEY=your_openai_api_key > .env
+```
+
+Then you can build the image with the following command. This will 
+automatically set up the environment and dependencies for CAMEL.
+
+```bash
+docker compose build
```

Comment:
any insight for this? thanks!
```
(camel-ai-py3.10) bash-3.2$ docker compose build
[+] Building 4.8s (10/11)                                                                   docker:desktop-linux
 => [camel internal] load build definition from Dockerfile                                                  0.0s
 => => transferring dockerfile: 466B                                                                        0.0s
 => [camel internal] load metadata for docker.io/library/python:3.10.10-bullseye                            1.0s
 => [camel internal] load .dockerignore                                                                     0.0s
 => => transferring context: 2B                                                                             0.0s
 => [camel 1/7] FROM docker.io/library/python:3.10.10-bullseye@sha256:44c9f73a82a6d95d85d4b768611ceb2c726f  0.0s
 => [camel internal] load build context                                                                     0.3s
 => => transferring context: 632.54kB                                                                       0.3s
 => CACHED [camel 2/7] RUN curl -sSL https://install.python-poetry.org | python3 - &&     ln -s /root/.loc  0.0s
 => CACHED [camel 3/7] RUN poetry config virtualenvs.create false                                           0.0s
 => CACHED [camel 4/7] COPY . /app/camel                                                                    0.0s
 => CACHED [camel 5/7] WORKDIR /app/camel                                                                   0.0s
 => ERROR [camel 6/7] RUN poetry install --with dev,docs -E all                                             3.5s
------                                                                                                           
 > [camel 6/7] RUN poetry install --with dev,docs -E all:                                                        
0.466 Skipping virtualenv creation, as specified in config file.                                                 
0.527 Installing dependencies from lock file                                                                     
1.148                                                                                                            
1.148 Package operations: 273 installs, 2 updates, 0 removals                                                    
1.148 
1.149   - Updating setuptools (65.5.1 -> 70.1.1)
1.151   - Updating wheel (0.40.0 -> 0.43.0)
2.170   - Installing certifi (2024.6.2)
2.171   - Installing charset-normalizer (3.3.2)
2.171   - Installing idna (3.7)
2.172   - Installing numpy (1.26.4)
2.177   - Installing nvidia-cublas-cu11 (11.10.3.66)
2.184   - Installing nvidia-cuda-nvrtc-cu11 (11.7.99)
2.190   - Installing nvidia-cuda-runtime-cu11 (11.7.99)
2.196   - Installing nvidia-cudnn-cu11 (8.5.0.96)
2.197   - Installing pycparser (2.22)
2.202   - Installing six (1.16.0)
2.203   - Installing typing-extensions (4.12.2)
2.206   - Installing urllib3 (2.2.2)
2.514 
2.514   RuntimeError
2.514 
2.514   Unable to find installation candidates for nvidia-cuda-nvrtc-cu11 (11.7.99)
2.514 
2.514   at ~/.local/share/pypoetry/venv/lib/python3.10/site-packages/poetry/installation/chooser.py:74 in choose_for
2.520        70│ 
2.520        71│             links.append(link)
2.520        72│ 
2.520        73│         if not links:
2.520     →  74│             raise RuntimeError(f"Unable to find installation candidates for {package}")
2.520        75│ 
2.520        76│         # Get the best link
2.521        77│         chosen = max(links, key=lambda link: self._sort_key(package, link))
2.521        78│ 
2.521 
2.521 Cannot install nvidia-cuda-nvrtc-cu11.
2.521 
2.522 
2.522   RuntimeError
2.522 
2.522   Unable to find installation candidates for nvidia-cublas-cu11 (11.10.3.66)
2.522 
2.522   at ~/.local/share/pypoetry/venv/lib/python3.10/site-packages/poetry/installation/chooser.py:74 in choose_for
2.525        70│ 
2.525        71│             links.append(link)
2.525        72│ 
2.525        73│         if not links:
2.525     →  74│             raise RuntimeError(f"Unable to find installation candidates for {package}")
2.526        75│ 
2.526        76│         # Get the best link
2.526        77│         chosen = max(links, key=lambda link: self._sort_key(package, link))
2.526        78│ 
2.526 
2.526 Cannot install nvidia-cublas-cu11.
2.526 
2.527 
2.527   RuntimeError
2.527 
2.527   Unable to find installation candidates for nvidia-cudnn-cu11 (8.5.0.96)
2.527 
2.528   at ~/.local/share/pypoetry/venv/lib/python3.10/site-packages/poetry/installation/chooser.py:74 in choose_for
2.532        70│ 
2.532        71│             links.append(link)
2.532        72│ 
2.532        73│         if not links:
2.532     →  74│             raise RuntimeError(f"Unable to find installation candidates for {package}")
2.532        75│ 
2.532        76│         # Get the best link
2.532        77│         chosen = max(links, key=lambda link: self._sort_key(package, link))
2.532        78│ 
2.532 
2.532 Cannot install nvidia-cudnn-cu11.
2.532 
2.533 
2.533   RuntimeError
2.534 
2.534   Unable to find installation candidates for nvidia-cuda-runtime-cu11 (11.7.99)
2.534 
2.534   at ~/.local/share/pypoetry/venv/lib/python3.10/site-packages/poetry/installation/chooser.py:74 in choose_for
2.538        70│ 
2.538        71│             links.append(link)
2.538        72│ 
2.538        73│         if not links:
2.538     →  74│             raise RuntimeError(f"Unable to find installation candidates for {package}")
2.538        75│ 
2.538        76│         # Get the best link
2.539        77│         chosen = max(links, key=lambda link: self._sort_key(package, link))
2.539        78│ 
2.539 
2.539 Cannot install nvidia-cuda-runtime-cu11.
2.539 
------
failed to solve: process "/bin/sh -c poetry install --with dev,docs -E all" did not complete successfully: exit code: 1
```

Reply:
This problem lies in that compatibility issue I mentioned last week, which is between torch and linux system of arm architecture. Maybe we should remove the torch dependency, as I don't see much usages of torch in our system (and it takes up a large space in our image), or we can add a condition in pyproject.toml to skip it on linux of arm architecture. I think we should raise a new pr to optimize and sort out our dependencies.